app/src/main/java/com/example/stretchy/features/executetraining/ui/ExecuteTrainingViewModel.kt

```diff
@@ -52,95 +62,241 @@ class ExecuteTrainingViewModel(val repository: Repository, val trainingId: Long)
         }
         _uiState.value = _uiState.value.copy(isLoading = true)
         viewModelScope.launch {
-            val trainingWithActivities = repository.getTrainingWithActivitiesById(trainingId)
-            getExercisesWithBreak(trainingWithActivities.activities).let { activitiesWithBreaks ->
-                activitiesWithBreaks.forEachIndexed { index, activity ->
-                    timer.setDuration(activity.duration)
-                    timer.flow.takeWhile { it >= 0 }.collect { currentSeconds ->
-                        when (activity.activityType) {
-                            ActivityType.STRETCH -> {
-                                handleStretch(
-                                    activity,
-                                    activitiesWithBreaks.getOrNull(index + 2)?.name,
-                                    currentSeconds,
-                                    index == 0
-                                )
-                            }
-                            ActivityType.BREAK -> {
-                                handleBreak(
-                                    activity,
-                                    activitiesWithBreaks.getOrNull(index + 1)?.name,
-                                    currentSeconds
-                                )
-                            }
-                            else -> {}
-                        }
+            trainingWithActivities = repository.getTrainingWithActivitiesById(trainingId)
+            initializeDisplayableList()
+
+            trainingWithActivities.activities.let { activitiesWithBreaks ->
+                while (true) {
+                    val currentActivity = activitiesWithBreaks[index]
+
+                    presetCurrentExercise(currentActivity)
+                    countDownTimeCurrentActivity(currentActivity)
```

Comment:
`countDownTimeCurrentActivity` I said that those function names are temporarey and you should change those.
Maybe some name that describes the feature in a business way. eg `strartExerciseTrainingFlow()`